### ISSUE TITLE
Change of style in matplotlib

### ISSUE THREAD

**Issue by neutrinoceros** [yt-project/yt#3514]
### Bug report

**Bug summary**

Matplotlib 3.4 came with a change in default math text font that affects how yt plots look by default. IMO the "classic" yt look is much nicer so I'd love that it stayed the default style as far as yt is concerned, with no regards to what matplotlib version is installed.

**Code for reproduction**

Here's a simple demo using yt 4.0 and matplotlib 3.4
```python
import yt
import matplotlib.pyplot as plt

def myplot():
    ds = yt.load_sample("IsolatedGalaxy")
    sp = yt.SlicePlot(ds, "z", ("gas", "density"))
    return sp

# default options with matplotlib 3.4
myplot().save("/tmp/defaults")
```

**Actual outcome**

![defaults_Slice_z_density](https://user-images.githubusercontent.com/14075922/133406771-e5d55c50-35ea-4afb-8dc1-a72d4e2b6e0d.png)

**Expected outcome**
This is the "classic yt" look that required no tweaking before Matplotlib 3.4
![classic_yt_Slice_z_density](https://user-images.githubusercontent.com/14075922/133406835-e8631925-d18e-4a35-aa8f-c219bf9b4193.png)

now it can be reproduced with any version of matplotlib with some additional effort on the user side.
```python
plt.rcParams["mathtext.fontset"] = "cm"
myplot().save("/tmp/classic_yt")
```
this is the simplest way to do it, but unfortunately it has side effects, as it won't just change the settings for that one yt plot, but globally change how matplotlib looks through out the session.

Side effects can be avoided using [a context manager](https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.rc_context.html)
```python
# with temporary settings (no side effects)
with plt.rc_context({"mathtext.fontset": "cm"}):
    myplot().save("/tmp/classic_yt")
```
but this technique is IMO too disruptive.

I suggest we could use such context manager internally to restore yt's default look with no tweaking on the user side. In order to preserve flexibility (users should still be able to set a different font), some design decisions would be needed.
What do other users and developers think ? 

**Version Information**
  * Operating System: OsX
  * Python Version: 3.9.5
  * yt version: 4.0.1
  * Other Libraries (if applicable): Matplotlib 3.4.2

**Comment by matthewturk** [comment]
I do like the old look, and the context manager approach seems right.

**Comment by chummels** [comment]
Agreed.  I like the old look as well.  Good job in spotting these changes!

**Comment by neutrinoceros** [comment]
Note: I want to research the exact PR that changed that upstream.

**Comment by neutrinoceros** [comment]
What a nice surprise this turned out to be (look who wrote issue 6518 !)

So apparently the change of default math font actually dates back to MPL 2.0, see https://matplotlib.org/2.0.0/users/whats_new.html#change-in-default-font

It is not clear why it only popped up in yt 5 years latter with MPL 3.4, but here is an issue that was _closed_ on MPL 3.4.0
- issue https://github.com/matplotlib/matplotlib/issues/6518
- PR https://github.com/matplotlib/matplotlib/pull/18145 (MPL 3.4.0)

So maybe there's what we need to know there, and it's possible that the change of behaviour wasn't intentional after all and could be considered a MPL bug. I'll investigate this further another time.

**Comment by neutrinoceros** [comment]
The change bisects to https://github.com/matplotlib/matplotlib/pull/18862

**Comment by chummels** [comment]
Interesting.  I don't understand why we're only just seeing it when it was merged almost a year ago.  Is it because that only made it into a recent MPL release that we started using?

**Comment by neutrinoceros** [comment]
The first release to contain the changing patch was 3.4.0, which wasn't released as long ago. At the time they were a couple worse incompatibilities to handle, so I set this one up aside. It was indeed more work to track it down and I'm a clearly not done yet.
Hopefully future releases will be less disruptive for us since we've started testing against the dev version of MPL

**Comment by neutrinoceros** [comment]
Follow up answer : turns out Nathan implemented custom parametrization inside yt to ensure style consistency without regards for matplotlib's defaults. Now because matplotlib introduced a new parameter with their 3.4 release, Nathan's implementation on the yt side needed a slight adjustment, but it turned out to be pretty easy once I figured out where it was. See #3520
